### ISSUE TITLE
happauge dual-hd usb 

### ISSUE THREAD

**Issue by unhookeddotnet** [hselasky/webcamd#3]
With webcamd-4.17.0.2, driver is only creating one dvb device. Not sure of status of device in v4l tree.

**Comment by hselasky** [comment]
Is this a regression issue?
Can you try to enable debugging and run webcamd in the foreground and collect the debug prints?

**Comment by hselasky** [comment]
Can you also provide related /var/log/messages printouts when the device is connected to a Linux box?

**Comment by unhookeddotnet** [comment]
I don't think it's a regression, wasn't supported before and there are linuxtv threads mentioning patches but I'm not sure if those were committed before you did your sync. What I do know is the tuner needs dvb-demod-si2168-b40-01.fw  I'll pull the stick from the shield this weekend and give it a run.

**Comment by unhookeddotnet** [comment]
http://unhooked.net/tv/webcamd is the output of /usr/local/sbin/webcamd -i 0 -d ugen1.3 -U webcamd -G webcamd -H
It looks like I had already compiled with debugging turned on. 
Someone elses dmesg is here: https://github.com/b-rad-NDi/Ubuntu-media-tree-kernel-builder/issues/31

**Comment by hselasky** [comment]
Hi,

INFO: i2c-131: Added multiplexed i2c bus 133
INFO: 131-000e: LG Electronics LGDT3306A successfully identified
DBG: 133-0060: failed=-6
DBG: 133-0060: failed=-6

Is it possible you could trace down where the failed -6 (ENXIO) prints come from and get a proper backtrace using gdb from ports?

--HPS

**Comment by hselasky** [comment]
Can you also add "-m si21xx.debug=1" argument for webcamd?

**Comment by unhookeddotnet** [comment]
my gdb skills are weak, but I've got http://unhooked.net/tv/webcamd and http://unhooked.net/tv/webcamd.gdb

**Comment by hselasky** [comment]
I'll try to give you some commands:
break printf
break puts
break fprintf
c  # continues the code
CTRL+C # breaks the code
bt # produces a backtrace

I need you to put a break on the print statements, then press "c" until you see error -6, then print "bt" and send that.

**Comment by unhookeddotnet** [comment]
It's not getting that far before the program exits, not sure what I'm doing
but it's going through the usb device detection and then exiting when I
either attach to the running process and issue a start, or when launching
from inside gdb.

But it doesn't look to me like it's loading firmware.

**Comment by hselasky** [comment]
Can you try again using the latest version of webcamd?

**Comment by unhookeddotnet** [comment]
building now, will test this weekend.

**Comment by unhookeddotnet** [comment]
Dmesg still reports it as Dec  5 11:02:12 its kernel: ugen0.8: <HCW 955D>
at usbus0

However, devd and webcam just did their thing.

w_scan -fa -c US -Z
w_scan version 20170107 (compiled for DVB API 5.10)
using settings for UNITED STATES
ATSC
VSB US/CA, DVB-T TW
scan type TERRCABLE_ATSC, channellist 1
output format w_scan XML tuning data
output charset 'ISO-8859-1'
Info: using DVB adapter auto detection.
        /dev/dvb/adapter0/frontend0 -> TERRCABLE_ATSC "LG Electronics
LGDT3306A VSB/QAM Frontend": good :-)
        /dev/dvb/adapter1/frontend0 -> TERRCABLE_ATSC "LG Electronics
LGDT3306A VSB/QAM Frontend": good :-)
Using TERRCABLE_ATSC frontend (adapter /dev/dvb/adapter0/frontend0)
-_-_-_-_ Getting frontend capabilities-_-_-_-_
Using DVB API 5.11
frontend 'LG Electronics LGDT3306A VSB/QAM Frontend' supports
INVERSION_AUTO
8VSB
QAM_64
QAM_256
FREQ (54.00MHz ... 858.00MHz)
-snip-
tune to: 8VSB     f=473000 kHz (0:0:0) (time: 03:48.344)
service is running. Channel number: 12:1. Name: 'KVOS-HD'
service is running. Channel number: 12:2. Name: 'Movies'
service is running. Channel number: 12:3. Name: 'MeTV'
service is running. Channel number: 12:4. Name: 'Decades'
service is running. Channel number: 12:5. Name: 'StartTV'
tune to: 8VSB     f=503000 kHz (0:0:0) (time: 03:49.913)
service is running. Channel number: 24:1. Name: 'KBCB'
service is running. Channel number: 24:2. Name: 'LEASE'
service is running. Channel number: 24:3. Name: 'QVC'
service is running. Channel number: 24:4. Name: 'JTV'
tune to: 8VSB     f=545000 kHz (0:0:0) (time: 03:51.848)
service is running. Channel number: 26:1. Name: 'Vancouver CBUFT-DT'
tune to: 8VSB     f=581000 kHz (0:0:0) (time: 03:53.594)
service is running. Channel number: 32:1. Name: 'CIVT'
tune to: 8VSB     f=599000 kHz (0:0:0) (time: 03:55.863)
service is running. Channel number: 2:1. Name: 'VANCOUVER CBUT-DT'
(time: 03:57.314) dumping lists (12 services)

**Comment by hselasky** [comment]
Hi,

Can you try the latest webcamd from FreeBSD ports? (v5.13.2.8)

Also try running webcamd from valgrind?

--HPS

**Comment by unhookeddotnet** [comment]
ugen0.8: <HCW 955D> at usbus0

[happ.log](https://github.com/hselasky/webcamd/files/7766702/happ.log)
[happ.txt](https://github.com/hselasky/webcamd/files/7766703/happ.txt)
sudo valgrind --log-file="/home/tom/happ.log" /usr/local/sbin/webcamd -i 0 -d ugen0.8 -U webcamd -G webcamd -H > happ.txt  2 > & 1

**Comment by hselasky** [comment]
Seems like it is working?

**Comment by unhookeddotnet** [comment]
> 
Yeah, still misidentified in dmesg, but webcamd detects everything correctly. I was able to build a channel map and watch atsc with vlc.
